src/main/java/org/jabref/gui/importer/actions/OpenDatabaseAction.java

```diff
@@ -184,49 +183,56 @@ else if (toRaise != null) {
+            BackgroundTask.wrap(() -> loadDatabase(file, raisePanel))
+                          .onSuccess(result -> OpenDatabaseAction.performPostOpenActions(result.getBasePanel(), result.getParserResult(), result.getDialogService()))
+                          .executeWith(Globals.TASK_EXECUTOR);
+        }
 
-            Globals.prefs.put(JabRefPreferences.WORKING_DIRECTORY, fileToLoad.getParent().toString());
+    }
 
-            if (BackupManager.checkForBackupFile(fileToLoad)) {
-                BackupUIManager.showRestoreBackupDialog(frame.getDialogService(), fileToLoad);
-            }
+    private ResultHelper loadDatabase(Path file, boolean raisePanel) {
+        Path fileToLoad = file.toAbsolutePath();
 
-            ParserResult result;
-            result = OpenDatabase.loadDatabase(fileToLoad.toString(),
-                    Globals.prefs.getImportFormatPreferences(), Globals.getFileUpdateMonitor());
+        frame.getDialogService().notify(Localization.lang("Opening") + ": '" + file + "'");
 
-            if (result.getDatabase().isShared()) {
-                try {
-                    new SharedDatabaseUIManager(frame).openSharedDatabaseFromParserResult(result);
-                } catch (SQLException | DatabaseNotSupportedException | InvalidDBMSConnectionPropertiesException |
-                        NotASharedDatabaseException e) {
-                    result.getDatabaseContext().clearDatabaseFile(); // do not open the original file
-                    result.getDatabase().clearSharedDatabaseID();
-                    LOGGER.error("Connection error", e);
+        Globals.prefs.put(JabRefPreferences.WORKING_DIRECTORY, fileToLoad.getParent().toString());
 
-                    frame.getDialogService().showErrorDialogAndWait(Localization.lang("Connection error"),
-                            e.getMessage() + "\n\n" + Localization.lang("A local copy will be opened."));
+        if (BackupManager.checkForBackupFile(fileToLoad)) {
+            BackupUIManager.showRestoreBackupDialog(frame.getDialogService(), fileToLoad);
+        }
 
-                }
+        ParserResult result;
+        result = OpenDatabase.loadDatabase(fileToLoad.toString(),
+                                           Globals.prefs.getImportFormatPreferences(), Globals.getFileUpdateMonitor());
+
+        if (result.getDatabase().isShared()) {
+            try {
+                new SharedDatabaseUIManager(frame).openSharedDatabaseFromParserResult(result);
+            } catch (SQLException | DatabaseNotSupportedException | InvalidDBMSConnectionPropertiesException |
+                     NotASharedDatabaseException e) {
+                result.getDatabaseContext().clearDatabaseFile(); // do not open the original file
+                result.getDatabase().clearSharedDatabaseID();
+                LOGGER.error("Connection error", e);
+
+                frame.getDialogService().showErrorDialogAndWait(Localization.lang("Connection error"),
```

Comment:
same here, not on JavaFX thread. Just rethrow the exception here (after cleaning up) and add `onException` handler to background task above